projects/lib/src/board/westernboard.cpp

```diff
@@ -631,17 +655,32 @@ bool WesternBoard::vSetFenString(const QStringList& fen)
 	m_sign = (side == Side::White) ? 1 : -1;
 	if (*token != "-")
 	{
-		setEnpassantSquare(squareIndex(*token));
+		int epSq = squareIndex(*token);
+		setEnpassantSquare(epSq);
 		if (m_enpassantSquare == 0)
 			return false;
 
+		Piece ownPawn(side, Pawn);
+		Piece opPawn(side.opposite(), Pawn);
+		int matchesOwn = 0;
+		int epTgt = 0;
+
+		for ( PawnStep pStep: pawnSteps )
```

Comment:
I would like to avoid deep copies as long it is possible and makes sense. I am working on that.